draft-ietf-jsonpath-base.md

```diff
@@ -555,6 +555,14 @@ A JSONPath query consists of a sequence of selectors. Valid selectors are
   * Filter selector `[?(<expr>)]`
   * Current item selector `@` (used in expressions)
 
+Note that processing the dot selector and string-valued index selectors
```

Comment:
Do we want to include comparisons in expressions?

Reply:
Right, as in filter selectors. Should definitely mention them.  Now, I haven't used them so I have a basic question. All our examples are of the form 

`$.foo == "bar"`

so the statement that you're comparing a string from the JSON to one from the JSONPath is still true.  But is it also legal to have a filter like `$.foo == $.bar`?  In that case you're comparing two strings from the JSON.  Which doesn't change the code points rule of course.

Reply:
Either side of `==` is a `comparable` which, for strings, could be either a string literal or a path beginning with `$` or `@`. So string comparisons include, for example:
* `"x" == "x"`
* `"x" == $.foo`
* `$.foo == "x"`
* `$.foo == $.bar`
* `@.foo == "x"`
* `@.foo == @.bar`
* `@.foo == $.bar`
* `$.bar == @.foo`

(There are also examples which would evaluate to false because a string is being compared to another type, such as `"true" == true`.)

Reply:
Despite `$.foo == $.bar` is valid in filters, it hardly makes sense. The current value `@` during iteration process plays no role.

Reply:
> Despite `$.foo == $.bar` is valid in filters, it hardly makes sense. The current value `@` during iteration process plays no role.

It could make some kind of sense, e.g. to select an array element if and only if two values in the document happen to be equal. Comparisons such as `"x" == "x"` make less sense because they are constant predicates, true or false.

Reply:
Actually, I can imagine something like `$.submittedBy == $.resourceOwner`

(actually more likely != but the point stands)

Will update the PR

Reply:
> Will update the PR

Time for that update, I'd say...